test/unit/webui/utils/login.spec.js

```diff
@@ -0,0 +1,123 @@
+import {Base64} from 'js-base64';
+import addHours from 'date-fns/add_hours';
+
+import {isTokenExpire, makeLogin} from '../../../../src/webui/utils/login';
+
+const generateTokenWithTimeRange = (limit = 0) => {
+  const payload = {
+    username: 'verdaccio',
+    exp: Number.parseInt(addHours(new Date(), limit).getTime() / 1000, 10)
+  };
+  return `xxxxxx.${Base64.encode(JSON.stringify(payload))}.xxxxxx`;
```

Comment:
`xxxx.${Base64.encode(JSON.stringify(payload))}.xxx` seems to be dupe as well, perhaps some helper module.